### PR TITLE
[FEAT] Cancel batches

### PR DESCRIPTION
**Pull request by Mildophin**
### Description

This PR adds the feature of cancelling a group of batches by providing a list of batch ids.

The documentation has been updated to display how to use the get batches method and cancel jobs method aswell.

### Checklist

- [ ] The title of the PR follows the right format: [{Label}] {Short Message}. Label examples: IMPROVEMENT, FIX, REFACTORING... Short message is about what your PR changes.

#### Versioning (PASQAL developers only)

- [ ] Update the version of pasqal-cloud in `_version.py` following the changes in your PR and by using [semantic versioning](https://semver.org/).

#### Documentation

- [ ] Update CHANGELOG.md with a description explaining briefly the changes to the users.

#### Tests

- [ ] Unit tests have been added or adjusted.
- [ ] Tests were run locally.

#### Internal tests pipeline (PASQAL developers only)
- [ ] Update and run the internal tests while targeting the branch of this PR.
 If your PR hasn't changed any functionality, it still needs to be validated against internal tests.

#### After updating the version (PASQAL developers only)

- [ ] Open a PR on the internal tests that updates the version used for the pasqal-cloud backward compatibility tests.

### CODE REVIEW

**Review by MatthieuMoreau0 — APPROVED** [review]

**Review by Augustinio — APPROVED** [review]
Should we add something to the README?

File: README.md [hunk 1]
```diff
@@ -177,6 +177,51 @@ for job in batch.ordered_jobs:
     print(f"job-id: {job.id}, status: {job.status}, result: {job.result}")
 ```
 
+### Get a list of batches
+
+It is possible to get all batches or a selection of batches with the `get_batches` method.
+This method uses a pagination system that you have to handle.
```
Comment (MatthieuMoreau0) [inline README.md:183]: "that you have to handle" reads weird to me.

Suggestion:
The method returns by default a page of the 100 most recent batches but can be configured using the `pagination_params` argument

**Review by MatthieuMoreau0 — APPROVED** [review]

File: pasqal_cloud/__init__.py [hunk 2]
```diff
@@ -359,6 +367,36 @@ def cancel_batch(self, id: str) -> Batch:
...
+            BatchCancellingError which spawns from a HTTPError
+
+        """
+
+        try:
+            response = self._client.cancel_batches(
+                batch_ids=batch_ids,
+            )
+        except HTTPError as e:
+            raise BatchCancellingError(e) from e
```
Comment (MohamedAminMallek) [inline pasqal_cloud/__init__.py:392]:
```suggestion
            raise BatchCancellingError("a user-friendly error message") from e
```
This provides a user-friendly message while retaining the original exception context for debugging.

Using `from e` links the original exception (e) as the cause of `BatchCancellingError`, so including e.message directly in BatchCancellingError isn’t necessary.

File: tests/test_batch.py [hunk 3]
```diff
@@ -866,3 +867,47 @@ def test_get_batches_raises_value_error_on_invalid_value_for_limit(
...
+        """
+        response = self.sdk.cancel_batches(batch_ids=batch_ids)
+        assert isinstance(response, BatchCancellationResponse)
+
+        for item in response.batches:
+            assert isinstance(item, Batch)
+
+        for k, v in response.errors.items():
+            assert isinstance(k, UUID)
+            assert isinstance(v, str)
```
Comment (Augustinio) [inline tests/test_batch.py:901]: If I'm not mistaken in your response fixture you're not including any errors so this is never really tested.

**Pull request merged**